### PR TITLE
mgr/dashboard: Change the text 'Login' to 'Log in'

### PR DESCRIPTION
**Pull request by nizamial09**
Screenshot from 2020-11-17 00-08-51

Fixes: https://tracker.ceph.com/issues/48253
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by s0nea** [comment]
jenkins test api

**Review by s0nea — APPROVED** [review]
While I'm really used to see `Login` instead of `Log in` I guess the latter is grammatically more correct :smile:

**Review by alfonsomthd — APPROVED** [review]

**Review by aaSharma14 — APPROVED** [review]

**Comment by epuertat** [comment]
> While I'm really used to see `Login` instead of `Log in` I guess the latter is grammatically more correct

@s0nea good one: the reason would be that "Login" is a noun, while "Log in" is a verb, and since all our buttons are actions ("Edit", "Create", "Delete", ...) it made more sense to use the action here too. What do you think?

**Comment by s0nea** [comment]
> @s0nea good one: the reason would be that "Login" is a noun, while "Log in" is a verb, and since all our buttons are actions ("Edit", "Create", "Delete", ...) it made more sense to use the action here too. What do you think?

@epuertat do you mean to leave `Login` as it is? - I thought `Edit`, `Create` and `Delete` are verbs... and `Log in` as well.

**Comment by epuertat** [comment]
> > @s0nea good one: the reason would be that "Login" is a noun, while "Log in" is a verb, and since all our buttons are actions ("Edit", "Create", "Delete", ...) it made more sense to use the action here too. What do you think?
> 
> @epuertat do you mean to leave `Login` as it is? - I thought `Edit`, `Create` and `Delete` are verbs... and `Log in` as well.

@s0nea just the opposite, `Log in` is my choice too! :D

File: src/pybind/mgr/dashboard/frontend/src/app/shared/enum/components.enum.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export enum Components {
-  auth = 'Login',
+  auth = 'Log in',
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/enum/components.enum.ts:2]: In this case, I think the rationale is just the opposite @nizamial09 : here everything is a noun (component), so Login (Component) makes more sense. Do we need this too?

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/enum/components.enum.ts]: I think no. I made the changes. Thanks @epuertat

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by epuertat** [comment]
> jenkins test dashboard

@nizamial09 In this case the e2e error is expected. The Login page e2e test is looking for a "Login" text in the page, so you would have to either update to "Log in" or use a better selector.

**Comment by tchaikov** [comment]
@nizamial09  ping?

**Comment by s0nea** [comment]
> @nizamial09 In this case the e2e error is expected. The Login page e2e test is looking for a "Login" text in the page, so you would have to either update to "Log in" or use a better selector.

With regards to 'better selector' I found https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements quite helpful. I had the same problem in my PR https://github.com/ceph/ceph/pull/37994.

**Comment by epuertat** [comment]
> > @nizamial09 In this case the e2e error is expected. The Login page e2e test is looking for a "Login" text in the page, so you would have to either update to "Log in" or use a better selector.
> 
> With regards to 'better selector' I found https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements quite helpful. I had the same problem in my PR #37994.

@s0nea Superb reference (would you mind adding that as a doc change in your PR?)! I recall having a [great discussion](https://tracker.ceph.com/issues/41781) with @p-se on this topic.

**Comment by nizamial09** [comment]
@tchaikov I'll fix the e2e and update it soon.

@s0nea @epuertat Thank you for the suggestions. Let me try it out locally.

**Comment by nizamial09** [comment]
@epuertat Updated this PR. Fixed the E2E

**Pull request merged**